### ISSUE TITLE
remove status and deleting published models

### ISSUE THREAD

**Issue by esgomezm** [bioimage-io/collection-bioimage-io#522]
Hey @oeway and @constantinpape 

Together with @FynnBe we were discussing what would be the best way to "remove", "block" or stop displaying models that we don't want to be there (but are already in zenodo), or models that we cannot update directly and we need to create a new zenodo doi + model card for them.

A case, for example, is [this one](https://bioimage.io/#/?tags=arabidopsis&id=10.5281%2Fzenodo.5749843), for which we will create a new model card, so the current model card should disappear from the web page. 

@FynnBe made an important point about not having broken links, not faking dois and that even if we remove the models from the webpage, the models will still be in Zenodo. On the other hand, I think that we should have a straightforward protocol to remove content that we don't want to be there (either because it's just deprecated information or errors, or because they contain undesired content). 

Would you recommend having a new "remove" status or what would be your proposal for such a protocol?

Thanks!

**Comment by oeway** [comment]
Do you mean make the model disappear from the website but keep the URL working?

**Comment by FynnBe** [comment]
> Do you mean make the model disappear from the website but keep the URL working?

essentially, yes

> Would you recommend having a new "remove" status or what would be your proposal for such a protocol?

This would be my suggestions: In addition to `blocked`, and `accepted` we can have a `removed` status. (either only for whole resources or also for specific versions analog to `blocked`/`accepted`)
A removed resource will stay in [collection.json](https://github.com/bioimage-io/collection-bioimage-io/blob/gh-pages/collection.json)/[collection.yaml](https://github.com/bioimage-io/collection-bioimage-io/blob/gh-pages/collection.yaml), but will be marked there with `"status": "removed"`. We can then exclude it from further CI testing and filter it out on bioimage.io

**Comment by oeway** [comment]
Ok, if the point is to keep the URL still working, I think we don't need to add another status. 

Here is what I would do, on the website, if we get a model id that is not available anymore in the model zoo, can render a dialog saying something like: "Oops, it seems the model has been removed, however you can find it here: <zenodo link> ".

**Comment by FynnBe** [comment]
how do you know it is not available anymore and not simply invalid in this scenario?

**Comment by oeway** [comment]
> how do you know it is not available anymore and not simply invalid in this scenario?

We can check it by sending a request to zenodo to check if there is a rdf.yaml file under that id.

**Comment by FynnBe** [comment]
> We can check it by sending a request to zenodo to check if there is a rdf.yaml file under that id.

We need to be able to remove an entry independently. Also in this particular use case as the duplication of record is mainly motivated by lack of access to the primary record.

How about we simply add a "removed" tag for this purpose. We only need to configure on the website to filter that one out per default, but overall very little change is needed.

**Comment by oeway** [comment]
> We need to be able to remove an entry independently. Also in this particular use case as the duplication of record is mainly motivated by lack of access to the primary record.

Why not just block the duplicated record, and it disappears from our collection. With the zenodo link, we can at least make sure the user will have access to the zenodo record? Do you mean you want that record still shown as a card?

> How about we simply add a "removed" tag for this purpose. We only need to configure on the website to filter that one out per default, but overall very little change is needed.

I think we need a bit more thinking here. What's difference compared to block that record? I guess you want to only make it invisible, but still be able to render from a link, right? In that case, don't you also want to show the successor record instead of the old one?

**Comment by FynnBe** [comment]
> Do you mean you want that record still shown as a card?

yes. I see this use case if a record was previously "accepted" and not always "blocked". -> links, documentation, etc.. keeps existing, while the record is no more discoverable per default.

### follow-up record(s)
There may or may not be one or more follow-up record(s). These we could hint at in the description. No need for a special mechanism. I do not want to automatically show the successor record; there is a difference to the new record. This should be a conscious step, one click away for convenience.
In the new record we can use the `parent` field to link the removed old record. 

### reusing animal nickname
I would be open to consider changing the animal nickname (removing it from the deleted and assigning it to the successor) on a case by case basis to ease use. While we should refrain from redirecting the (doi based) record ids, I think we can make this educated guess for the animal nicknames, for which we give fewer guarantees.

**Comment by esgomezm** [comment]
pointing @fjug and @akreshuk here as they can see the discussion related to the models' graveyard

**Comment by fjug** [comment]
I think many ideas hear sounds good and could lead to a suitable solution. Still, I propose you guys tinker a bit more and we then discuss on a Wednesday meeting (I will try to join tomorrow, but will be at a conference somewhere in .de).
Just to poor a bit of oil in the fire: why is the status not a list of tupels (status, date), then the latest is the current one and we fixed the status history issue… what do you think?
And: we NEED a graveyeard in one way or another… that is just too cool to not have it! :)

**Comment by FynnBe** [comment]
> why is the status not a list of tupels (status, date),

the status history is already included in the git history. If we need it to be more available, we could add it like that.
But do we need the status history handy? After a long status history a resource is either removed or accepted. I think reviving resources is going to be very very rare

![image](https://user-images.githubusercontent.com/15139589/222086538-11cb71fc-d110-4451-b989-76e23a111ded.png)

pending: the record does not exist in the collection repo, only on zenodo -> It has no status yet

**Comment by akreshuk** [comment]
Better/gentler name for the graveyard: https://en.wiktionary.org/wiki/farm_upstate
And we do need a status, for now "retired" sounds reasonable.

**Comment by oeway** [comment]
Sounds good!

Using `retired` is actually better, the model zoo won't kill any animals, and we won't even let them die ;)
